fpmsyncd/routesync.cpp

```diff
@@ -30,6 +54,466 @@ RouteSync::RouteSync(RedisPipeline *pipeline) :
+
+            for (;;) 
+            {
+                uint16_t encap = 0;
+                if (len < (int)sizeof(*rtnh)
+                    || rtnh->rtnh_len > len)
+                    break;
+
+                gate = 0;
+                if (rtnh->rtnh_len > sizeof(*rtnh)) 
+                {
+                    memset(subtb, 0, sizeof(subtb));
+
+                    netlink_parse_rtattr(subtb, RTA_MAX, RTNH_DATA(rtnh),
+                                          (int)(rtnh->rtnh_len - sizeof(*rtnh)));
+
+                    if (subtb[RTA_GATEWAY])
+                        gate = RTA_DATA(subtb[RTA_GATEWAY]);
+
+                    if (gate)
+                    {
+                        if (RTA_PAYLOAD(subtb[RTA_GATEWAY]) <= IPV4_MAX_BYTE)
+                        {
+                            memcpy(gateaddr, gate, IPV4_MAX_BYTE);
+                            gw_af = AF_INET;
+                        }
+                        else
+                        {
+                            memcpy(ipv6_address.s6_addr, gate, IPV6_MAX_BYTE);
+                            gw_af = AF_INET6;                    
+                        }
+                    }
+                    
+                    if(gw_af == AF_INET6)
+                    {
+                        if (IN6_IS_ADDR_V4MAPPED(&ipv6_address))
+                        {
+                            memcpy(gateaddr, (ipv6_address.s6_addr+12), IPV4_MAX_BYTE);
+                            gw_af = AF_INET;
+                        }
+                        else
+                        {
+                            SWSS_LOG_NOTICE("IPv6 tunnel nexthop not supported Nexthop:%s encap:%d encap_value:%d",
+                                            inet_ntop(gw_af, ipv6_address.s6_addr, nexthopaddr, MAX_ADDR_SIZE), encap, encap_value);
+                            return false;
+                        }
+                    }
+                    
+                    inet_ntop(gw_af, gateaddr, nexthopaddr, MAX_ADDR_SIZE);
+
+                    if (subtb[RTA_ENCAP_TYPE])
+                        encap = *(uint16_t *)RTA_DATA(subtb[RTA_ENCAP_TYPE]);
+
+                    if (subtb[RTA_ENCAP] && subtb[RTA_ENCAP_TYPE]
+                        && (*(uint16_t *)RTA_DATA(subtb[RTA_ENCAP_TYPE]) == NH_ENCAP_VXLAN))
+                    {
+                        parse_encap(subtb[RTA_ENCAP], encap_value, rmac, vlan);
+                    }
+                    SWSS_LOG_DEBUG("Multipath Nexthop:%s encap:%d encap_value:%d rmac:%s vlan:%d",
+                                    nexthopaddr, encap, encap_value, rmac.c_str(), vlan);
+
+                    if (encap_value == 0 || vlan == 0
+                        || MacAddress(rmac) == MacAddress("00:00:00:00:00:00"))
+                        return false;
+
+                    if(ecmp_count)
+                    {
+                        getEvpnNextHopSep(nexthops, vni_list, mac_list, intf_list);
+                    }
+
```

Comment:
Add this only if the message has RTA_GATEWAY??

Reply:
Update the code based on the comment